src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs

```diff
@@ -63,20 +64,8 @@ public class MongoDatabaseWrapper : Database
     /// <returns>Number of documents affected during saving changes.</returns>
     public override int SaveChanges(IList<IUpdateEntry> entries)
     {
-        int docsAffected = 0;
-
-        // TODO: Consider splitting the changes up by type and using the Mongo bulk APIs instead
-
-        foreach (var entry in GetAllChangedRootEntries(entries))
-        {
-            // TODO: Exception handling
-            if (Save(entry))
-            {
-                docsAffected++;
-            }
-        }
-
-        return docsAffected;
+        var updates = ConvertUpdateEntriesToMongoUpdates(entries);
+        return (int)SaveMongoUpdates(updates);
```

Comment:
At a high level saving changes is merely a matter of converting the EF Core update entries to MongoDB updates and running a bulk update.